File: sdkwork-backend-sdk-typescript/src/types/invitation-relation-form.ts
/** Invitation relation creation form */
export interface InvitationRelationForm {
  /** Inviter user ID */
  inviterUserId: number;
  /** Invitee user ID */
  inviteeUserId: number;
  /** Invite code */
  inviteCode: string;
  /** Used time */
  usedTime?: string;
  /** Relation level */
  relationLevel: number;
  /** Reward status */
  rewardStatus?: 'PENDING' | 'SUCCESS' | 'FAILED' | 'CANCELED';
  /** Reward amount */
  rewardAmount?: number;
  /** Reward type */
  rewardType?: 'POINTS' | 'CASH' | 'COUPON' | 'VIP_DAYS' | 'DISCOUNT' | 'FREE_SHIPPING' | 'UPGRADE' | 'GIFT' | 'LOTTERY_CHANCE' | 'STORAGE' | 'FEATURE_UNLOCK' | 'CREDIT' | 'CONTENT' | 'MIXED';
}

File: sdkwork-backend-sdk-go/types/invitation_relation_form.go
package types

// Invitation relation creation form
type InvitationRelationForm struct {
	InviterUserId int `json:"inviterUserId"`
	InviteeUserId int `json:"inviteeUserId"`
	InviteCode string `json:"inviteCode"`
	UsedTime string `json:"usedTime"`
	RelationLevel int `json:"relationLevel"`
	RewardStatus string `json:"rewardStatus"`
	RewardAmount float64 `json:"rewardAmount"`
	RewardType string `json:"rewardType"`
}
